### PR TITLE
HomeActivity setup.

### PR DESCRIPTION
**Pull request by ShubhamWorks02**
## Description

This PR introduces the `feature/home_activity_ui` branch into the `develop` branch. The primary focus of this branch has been developing HomeActivity for this app.

### Changes Made

1. **Bottom Navigation View**: Implemented a bottom navigation view in the home activity. This allows users to seamlessly navigate between different sections of the app.

2. **Fragment Container**: Added a flexible fragment container within the home activity layout. This container will hold various fragments, ensuring a clean and modular architecture.

3. **Navigation Component Integration**: Integrated the Android Navigation Component with the bottom navigation view and fragment container. This allows for smooth navigation transitions and simplified backstack management.

### Checklist

- [x] Code follows the project's coding conventions.
- [x] No conflicts with the latest `develop` branch.

### CODE REVIEW

File: app/src/main/java/com/nearbynest/app/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+package com.nearbynest.app.ui.home
+
+import com.nearbynest.app.ui.base.BaseViewModel
+import timber.log.Timber
+
+class HomeViewModel : BaseViewModel() {
+    init {
+        Timber.e("Initialized")
```
Comment (prem-p-simform) [inline app/src/main/java/com/nearbynest/app/ui/home/HomeViewModel.kt]: Why `e` ?

File: app/src/main/java/com/nearbynest/app/ui/list/ListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package com.nearbynest.app.ui.list
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.nearbynest.app.R
+
+class ListFragment : Fragment() {
```
Comment (prem-p-simform) [inline app/src/main/java/com/nearbynest/app/ui/list/ListFragment.kt]: Use `BaseFragment` and `ListFragment` is not the best name.

File: app/src/main/java/com/nearbynest/app/ui/map/MapFragment.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package com.nearbynest.app.ui.map
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.nearbynest.app.R
+
+class MapFragment : Fragment() {
```
Comment (prem-p-simform) [inline app/src/main/java/com/nearbynest/app/ui/map/MapFragment.kt]: User base classes please.

File: app/src/main/res/navigation/nav_graph.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    android:id="@+id/nav_graph"
+    app:startDestination="@id/listFragment">
+
+    <fragment
+        android:id="@+id/mapFragment"
+        android:name="com.nearbynest.app.ui.map.MapFragment"
+        android:label="fragment_map"
+        tools:layout="@layout/fragment_map" />
+    <fragment
+        android:id="@+id/listFragment"
```
Comment (prem-p-simform) [inline app/src/main/res/navigation/nav_graph.xml]: Id should be self descriptive. List is very generic word.

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -6,7 +6,45 @@
     <dimen name="space_medium">20dp</dimen>
     <dimen name="space_large">32dp</dimen>
     <dimen name="space_extra_large">64dp</dimen>
-
+    <dimen name="dp_0">0dp</dimen>
```
Comment (prem-p-simform) [inline app/src/main/res/values/dimens.xml]: This is unnecessary work. Take a reference of above names of dimens. Write some practical names which you can relate to.

**Review by prem-p-simform — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/nearbynest/app/ui/home/HomeViewModel.kt (hunk 1 above)
Comment (ShubhamWorks02) [reply app/src/main/java/com/nearbynest/app/ui/home/HomeViewModel.kt]: I understood `e` is used for errors.
Replaced it with 'd'.

File: app/src/main/java/com/nearbynest/app/ui/list/ListFragment.kt (hunk 2 above)
Comment (ShubhamWorks02) [reply app/src/main/java/com/nearbynest/app/ui/list/ListFragment.kt]: Made necessary changes which uses BaseFragment in respective Fragment.

File: app/src/main/java/com/nearbynest/app/ui/map/MapFragment.kt (hunk 3 above)
Comment (ShubhamWorks02) [reply app/src/main/java/com/nearbynest/app/ui/map/MapFragment.kt]: Thanks for feedback.
Applied BaseFragment for each fragment used.

File: app/src/main/res/navigation/nav_graph.xml (hunk 4 above)
Comment (ShubhamWorks02) [reply app/src/main/res/navigation/nav_graph.xml]: Yes , List is bit more generic.
Used `propertyListFragment` instead `listFragment`.

File: app/src/main/res/values/dimens.xml (hunk 5 above)
Comment (ShubhamWorks02) [reply app/src/main/res/values/dimens.xml]: Removed that Size specific named format.
Will use practical descriptive names from now on as you suggested.

**Review by prem-p-simform — APPROVED** [review]

**Pull request merged**